### ISSUE TITLE
Spyder getting stuck and running a 'Hello World' script indefinitely

### ISSUE THREAD

**Issue by JackHenry281** [spyder-ide/spyder#16609]
## Problem Description
Spyder gets stuck running scripts. I am also getting a 'Completion and linting in the edit.....' error. I have reinstalled conda (miniconda), I have disabled the windows firewall and real time protection, I have run the anaconda prompt as an administrator, this has not fixed the issue. I think adding matplotlib to the environment is causing the issues.

### What steps reproduce the problem?

1. conda create -n broken1 python numpy pandas xarray spyder matplotlib
2. conda activate atmoss
3. spyder
4. create a simple classic print('Hello World') script
5. run the script a few times. by ~10 times it should get stuck

What seems to be working:
1. conda create -n working1 python numpy pandas xarray spyder

If I leave the bugged script running eventually spyder gives the 'Completion and linting in the edit.....' error. I have tried disabling the windows firewall and real time protection but I'm still getting the same issue.
![image](https://user-images.githubusercontent.com/46002710/137891527-5a5247c4-dc0a-4e32-9f29-5469fa4c3e54.png)

## Versions

* Spyder version: 5.0.5
* Python version: 3.9.7
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: Windows 10 

### Dependencies

```
* Spyder version: 5.0.5 None
* Python version: 3.9.7 64-bit
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10
```

**Comment by ccordoba12** [comment]
Hey @JackHenry281, thanks for reporting. You said

> I think adding matplotlib to the environment is causing the issues.

This certainly could be the problem because we had some bugs when setting the Matplotlib backend.

To test, please close Spyder, open the Anaconda Prompt, run there

    spyder --reset

and try again.

Let us know if it works for you.

**Comment by JackHenry281** [comment]
Hi,

Thanks for the help. I think resetting spyder has fixed the issue. I also spoke with a colleague who, by coincidence, had the same issue. They also found resetting spyder fixed the issue.

Sorry for missing that step when I was trouble shooting by myself :)

Best wishes, Jack

**Comment by ccordoba12** [comment]
No worries, it's fine. For the future, please avoid setting the Tk backend from our Preferences. That's what causes this problem and what we'll fix in our 5.2.0 version, to be released in November.
